internal/controller/controllers/hypershiftserviceconfig_controller.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	_ "embed"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// AgentServiceConfigReconciler reconciles a AgentServiceConfig object
+type HypershiftServiceConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logrus.FieldLogger
+
+	// Namespace the operator is running in
+	Namespace string
```

Comment:
Why do we need this?

Reply:
This is similar to the one used in ASC: https://github.com/openshift/assisted-service/blob/4ba9c5001e942101ac3f25df00e2bccb78592610/internal/controller/controllers/agentserviceconfig_controller.go#L110
We'll need it for stuff deployed on the hub cluster (ServiceAccount/Roles/etc). As opposed to entities deployed on the spoke clusters using a different namespace from the new CRD.

Reply:
RBAC for the operator will be installed when the operator is installed.

I don't understand what this operator would use in its own namespace.
Shouldn't all the work be done in the namespace of the CR?

Reply:
For webhooks as an example, we need the agentinstalladmission service deployed with the operator namespace. This is currently done by the ASC controller. So if we require an agent CR to be deployed as a prerequisite, it can still be handled by ASC. But if we want the new CRD to handle it, then it needs to have the operator ns. Anyway, we can probably just keep it simple for now and require an ASC, then handle it differently later if we want.

Reply:
It's not just a matter of keeping things simple. The hypershift reconciliation loop needs to provision and monitor resources on both the management cluster and the hypershift cluster. 
The Namespace variable points to the the namespace of the operator on the management cluster. 

In addition, We wouldn't like the ASC to handle any of the new provisioning logic. In the case of RBAC and WebHooks, that logic differs from the one in the ASC reconcile loop.

Reply:
> The hypershift reconciliation loop needs to provision and monitor resources on both the management cluster and the hypershift cluster.

Yes, but nearly all of the resources this operator deals with on the management cluster will be in the same namespace as the `HypershiftServiceConfig` resource, not the namespace as the operator.

The webhooks are an interesting case. They don't seem related to an individual instance of assisted-service, but to all the CRDs of a particular type in the cluster. So what's the idea, are we going to reconcile those for every different HSC? I'm assuming their configuration is just static and won't change with any of the data in an individual HSC, right?

Either way, while this is an interesting conversation, this `Namespace` field isn't being used in this PR, so just remove it until it's needed.

Reply:
Done